bokeh/models/widgets/inputs.py

```diff
@@ -43,6 +43,10 @@ class TextInput(InputWidget):
     A callback to run in the browser whenever the user unfocuses the TextInput
     widget by hitting Enter or clicking outside of the text box area.
     """)
+    
+    placeholder = String(default="", help="""
+    Placeholder for empty input field
+    """)
```

Comment:
@bryevdv 
The placeholder is used for empty input fields.
See here: http://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_input_placeholder
I would like to give the user information about the format that the field should have. E.g. for datetime input the placeholder is `2016-12-12 12:00:03`. The user now has the information which format the input should have. An therefore the `pattern` would be a good improvement too.